crates/client/rpc/src/methods/trace/trace_implementation.rs

```diff
@@ -25,19 +21,21 @@ use sp_blockchain::HeaderBackend;
 use sp_runtime::traits::Block as BlockT;
 use starknet_api::api_core::{ClassHash, ContractAddress};
 use starknet_core::types::{
-    BlockId, BroadcastedTransaction, DeclareTransactionTrace, DeployAccountTransactionTrace, ExecuteInvocation,
-    FeeEstimate, InvokeTransactionTrace, L1HandlerTransactionTrace, RevertedInvocation, SimulatedTransaction,
-    SimulationFlag, TransactionTrace, TransactionTraceWithHash,
+    BlockId, DeclareTransactionTrace, DeployAccountTransactionTrace, ExecuteInvocation, InvokeTransactionTrace,
+    L1HandlerTransactionTrace, RevertedInvocation, TransactionTrace, TransactionTraceWithHash,
 };
 use starknet_ff::FieldElement;
-use thiserror::Error;
 
+use super::lib::*;
 use crate::errors::StarknetRpcApiError;
-use crate::Starknet;
+use crate::utils::get_block_by_block_hash;
+use crate::{Starknet, StarknetReadRpcApiServer};
 
-#[async_trait]
 #[allow(unused_variables)]
-impl<A, B, BE, G, C, P, H> StarknetTraceRpcApiServer for Starknet<A, B, BE, G, C, P, H>
+pub async fn trace_block_transactions<A, B, BE, G, C, P, H>(
```

Comment:
Perhaps use a different file for `trace_block_transactions` and `trance_transactions`, following the same pattern as the rest of the refactoring.

Reply:
Yes but i implemented those methods in madara and separate these methods is not really necessary bc they use exactly the same pattern so its not very relevant to separate them according to me (like the difference between these 2 is only a empty vector)